app/worker/image-worker.coffee

```diff
@@ -0,0 +1,20 @@
+importScripts 'jimp.min.js'
+
+MAX_SIZE = 1448
+MAX_FILE_SIZE = 5 * 1024
+
+self.addEventListener 'message', (e) ->
+  data = e.data
+
+  Jimp.read(data.message).then (image) ->
+
+    if image.bitmap.width > MAX_SIZE or image.bitmap.height > MAX_SIZE
+      image.scaleToFit MAX_SIZE, MAX_SIZE
+
+    if image.bitmap.data.length > MAX_FILE_SIZE
+      image.quality 75
```

Comment:
maybe we should enable this rule: http://eslint.org/docs/rules/no-magic-numbers or at least start with the `warning` :)